tests/integration_tests/modules/test_version_change.py

```diff
@@ -0,0 +1,55 @@
+from pathlib import Path
+
+from tests.integration_tests.instances import IntegrationInstance
+from tests.integration_tests.util import ASSETS_DIR
+
+
+PICKLE_PATH = Path('/var/lib/cloud/instance/obj.pkl')
+TEST_PICKLE = ASSETS_DIR / 'test_version_change.pkl'
+
+
+def _assert_no_pickle_problems(log):
+    assert 'Failed loading pickled blob' not in log
+    assert 'Traceback' not in log
+    assert 'WARN' not in log
+
+
+def test_reboot_without_version_change(client: IntegrationInstance):
+    log = client.read_from_file('/var/log/cloud-init.log')
+    assert 'Python version change detected' not in log
+    assert 'Could not determine Python version used to write cache' not in log
```

Comment:
I think this needs to now be the following right throughout?
```suggestion
    assert 'Cache compatibility status is currently unknown.' not in log    
```